### PR TITLE
movies view model, movies state added

### PR DESCRIPTION
**Pull request by hin1129**
state, view model added

### CODE REVIEW

File: app/src/main/java/com/example/moviesdbproject/viewmodel/MovieState.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.example.moviesdbproject.viewmodel
+
+sealed interface MovieState{
+    object LOADING : MovieState
+    class SUCCESS<T>(response: T): MovieState
```
Comment (inaki92) [inline app/src/main/java/com/example/moviesdbproject/viewmodel/MovieState.kt:5]: This need to have val in order to be used in the code

File: app/src/main/java/com/example/moviesdbproject/viewmodel/MovieViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        viewModelScope.launch(coroutineDispatcher) {
+            try {
+                val response = movieRepository.getPlayNowMovies()
+                if (response.isSuccessful){
+                    response.body() ?.let {
+                        _moviesLiveData.postValue(MovieState.SUCCESS(it))
+                    }?: throw Exception("playing now movie no response")
+                }
+                else {
+                    throw Exception("playing now movie unsuccessful")
```
Comment (inaki92) [inline app/src/main/java/com/example/moviesdbproject/viewmodel/MovieViewModel.kt:33]: Use capital at the beginning

**Review by OlegAndroidDev — COMMENTED** [review]
LGTM

**Review by inaki92 — APPROVED** [review]
Please take a look at the comment

**Review by OlegAndroidDev — APPROVED** [review]

**Pull request merged**